app/controllers/katello/concerns/hosts_controller_extensions.rb

```diff
@@ -23,6 +26,35 @@ def puppet_environment_for_content_view
           cvpe = Katello::ContentViewPuppetEnvironment.where(:environment_id => environment, :content_view_version_id => version).first
           render :json => cvpe.nil? ? nil : {:name => cvpe.puppet_environment.name, :id => cvpe.puppet_environment.id}
         end
+
+        def content_hosts
+          respond_to do |format|
+            format.csv do
+              @hosts = resource_base_with_search.where(organization_id: params[:organization_id])
+                         .preload(:subscription_status_object, :content_facet, :subscription_facet)
+              csv_response(@hosts,
```

Comment:
There is no view, the data is streamed directly one row at a time by the CSV exporter to the client. Since reports can sometimes be very large, we don't want to block the download until they are generated completely.
This is only complex because of the amount of columns and relations this table has, other tables are much simpler (you can see some examples in foreman core).